### PR TITLE
Add a warning about keeping min.js up to date

### LINKED ISSUE #43: Minify js as part of build
Currently we rely on people manually minifying the file, we should add a git hook or a build.cmd or something which minifies the js automatically so the `.js` and `.min.js` don't get out of step.

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #43. Added a warning about keeping min.js up to date, updated min.js and made gulp keep the license.

### CODE REVIEW

File: jquery.validate.unobtrusive.js [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
     function escapeAttributeValue(value) {
         // As mentioned on http://api.jquery.com/category/selectors/
-        return value.replace(/([!"#$%&'()*+,./:;<=>?@\[\\\]^`{|}~])/g, "\\\\$1");
+        return value.replace(/([!"#$%&'()*+,./:;<=>?@\[\\\]^`{|}~])/g, "\\$1");
```
Comment (Eilon) [inline jquery.validate.unobtrusive.js]: And we're sure that reverting this line doesn't break anything? What was it trying to fix, and why did we think this fix was correct at the time?

Comment (ryanbrandenburg) [inline jquery.validate.unobtrusive.js]: It was trying to fix #33. The actual problem there was what I commented a little bit ago, that his data-val-for was subtly wrong. My best bet for why we thought it was correct at the time is basically that I was testing the wrong stuff. I think I had hand-wrote his html into my test app and in doing so inadvertently fixed the typo, and my test app was probably also serving the un-updated *.min.js.

Lessons for me: 1. Always repro and fix in the same setup 2. As much as possible, verify that the code I changed is actually getting run.

Comment (Eilon) [inline jquery.validate.unobtrusive.js]: Understood!

**Comment by Eilon** [comment]
:shipit:

File: README.md [hunk 2]
```diff
@@ -5,3 +5,4 @@ The jQuery Unobtrusive Validation library complements jQuery Validation by addin
 
 This project is part of ASP.NET Core. You can find samples, documentation and getting started instructions for ASP.NET Core at the [Home](https://github.com/aspnet/home) repo.
 
+`jquery.validate.unobtrusive.min.js` must be kept up to date with `jquery.validate.unobtrusive.js`, this can easily be done by running `gulp` before commiting.
```
Comment (Eilon) [inline README.md]: `committing`

**Pull request merged**